### PR TITLE
Add middleware settings in configuration

### LINKED ISSUE #207: Allow developers write their own interceptors
The current PHP package has a hidden feature - interceptors for grpc requests. 
As seen here: `\Google\Ads\GoogleAds\Lib\V1\ServiceClientFactoryTrait::getGoogleAdsClientOptions`
One application for them is logging (which is already implemented).

For a regular developer, it is not easy to use them and there is no documentation for it available. I feel there is a HUGE benefit in making them available to the general public. For example, I used them to create a VCR layer which would record and playback of all requests and responses (for purposes of testing).

The problem is that `\Google\Ads\GoogleAds\Lib\V1\GoogleAdsClient` is final and doesn't let you customize interceptors.

What do you think?

### PR DESCRIPTION
**Pull request by PierrickVoulet**
Fixes #207

### CODE REVIEW

**Comment by PierrickVoulet** [comment]
@fiboknacky I only implemented the change for V8 as of now. I will replicate for V7 when it looks good to you.

File: src/Google/Ads/GoogleAds/Lib/ConfigurationTrait.php [hunk 1]
```diff
@@ -35,6 +35,8 @@ trait ConfigurationTrait
     private $logLevel;
     private $proxy;
     private $transport;
+    private $unaryMiddlewares;
```
Comment (fiboknacky) [inline src/Google/Ads/GoogleAds/Lib/ConfigurationTrait.php:38]: I believe we use this class to hold things related to configuration (e.g., those we set in the google_ads_php.ini).
How do you plan to support providing them in the config file?

Comment (PierrickVoulet) [reply src/Google/Ads/GoogleAds/Lib/ConfigurationTrait.php:38]: There is no plans to support this in the config file.

We could do it in the future if we decide to support config file in PHP format to complement or replace the current INI format but this is out of scope of this PR.

Comment (fiboknacky) [reply src/Google/Ads/GoogleAds/Lib/ConfigurationTrait.php:38]: Can you somehow state this anywhere like in the config file itself or as a comment in here?
We've had the 1-1 mapping between the configuration and the config file for years. Deviating from this convention should at least has some explanation.

Comment (PierrickVoulet) [reply src/Google/Ads/GoogleAds/Lib/ConfigurationTrait.php:41]: My plan was to document this in our library doc (DevSite) but I agree that it should be stated in the source code as well, done!

**Review by fiboknacky — APPROVED** [review]

**Pull request merged**